yb-voyager/src/srcdb/yugabytedb.go

```diff
@@ -523,17 +551,24 @@ func (yb *YugabyteDB) FilterUnsupportedTables(migrationUUID uuid.UUID, tableList
 		if len(userDefinedTypes) == 0 {
 			continue
 		}
+		enumTypes := yb.getAllEnumTypesInSchema(sname)
 		tableColumnArrayTypes := yb.getTypesOfAllArraysInATable(sname, tname)
 		if len(tableColumnArrayTypes) == 0 {
 			continue
 		}
 
-		// If any of the data types of the arrays are in the enum types then add the table to the unsupported tables list
+		// If any of the data types of the arrays are in the user defined types then add the table to the unsupported tables list
 		// udt_type/data_type looks like status_enum[] whereas enum_type looks like status_enum
 	outer:
 		for _, arrayType := range tableColumnArrayTypes {
+			baseType := strings.TrimLeft(arrayType, "_")
 			for _, udt := range userDefinedTypes {
-				if strings.EqualFold(strings.TrimLeft(arrayType, "_"), udt) {
+				if strings.EqualFold(baseType, udt) {
+					// Check if this is an array of enum and we're using logical connector
+					if slices.Contains(enumTypes, baseType) && !yb.source.IsYBGrpcConnector {
+						// Array of enums are supported with logical connector, skip to next array column
+						continue outer
+					}
```

Comment:
Instead of this, How about this?
```
udtTypes := getUDTTypes() // only UDT without ENUMs
enumTypes := getEnumTypes()
unsupportedTableTypes := udtTypes 
if yb.source.IsYBGrpcConnector {
  unsupportedTableTypes = append( unsupportedTableTypes, enumTypes...)
}
```
then just use `unsupportedTableTypes`
```
for _, unsupportedType := range unsupportedTableTypes {
				if strings.EqualFold(baseType, unsupportedType) {
```

Reply:
Makes sense.